### DISCUSSION TITLE
Calgary, Canada Results

### DISCUSSION THREAD

**Discussion by brianlow** [bemasher/rtlamr#217] [General]
Some notes for anyone in Calgary or possibly Alberta

### Over the past couple days I've received messages for:
* ~110 unique meter IDs
* `SCM`, `SCM+` and `IDM` message types
* commodities:
  * gas: `12`, `156`
  * water: `03`, `13`, `171`
  * electricity: `05`, `07`
* have only received messages on rtlamr's default frequency (912 MHz I believe)
* used [rtl_433](https://github.com/merbanan/rtl_433) to search other frequencies but found no additional messages (315, 433, 868, 345, 910, 913, 918, 920, 923, 926, 950, 953, 956, 960 MHz), hopped between frequencies every 20 second, listening for I think 4 hours

### Gas meters
* I suspect almost all residential gas meters in Calgary transmit continuously 
* ATCO is/was experimenting with [meter reading from the air](https://www.atco.com/en-ca/about-us/stories/aerial-metre-reading-program.html)

### Water meters
* Some good info: [City of Calgary 2018 Metering Review](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwi_2LK706P0AhWG754KHYjxBWY4ChAWegQIEhAB&url=https%3A%2F%2Fpub-calgary.escribemeetings.com%2Ffilestream.ashx%3FDocumentId%3D36254&usg=AOvVaw1KkRDRKEo-JtG9W68ezHlz)
* City is using ITron 50W, 60W and 100W transmitters
* The 50W are likely all in "Wake Up" mode where they only transmit when a meter reader transmits a wake up signal. The above report doesn't mention battery replacement and in Wake Up mode the battery should last 20 years which is the lifetime of the meter. My meter (Commodity Type 03) only transmitted one burst over 2 days. I received 19 messages with type 03 around 7:30am one day. Really seems like triggered by a meter reader driving through the neighboorhood.
* I have received several thousand messages with commodity types 13 and 171 so I think there are some meters that transmit continuously. Maybe these are the newer 100W
* Managed by Enmax

### Power meters
* Based internet research, most meters do not have transmitters
* I have received thousands of power meter messages so perhaps newer meters transmit
* Managed by Enmax

### My house
* **Gas**: transmits every 15 seconds, Sensus (?) meter, FCC ID: `EWQ100GDLAN`
* **Water**: transmitted one burst of 13 messages, Elster/AMCO Scancoder meter, with ITron 50W transmitter, FCC ID: `EO940GN-1`
* **Power**: nothing, no FCC ID label
* House built 2004, inner cityish, so wide age range of houses nearby

### My setup
`Nooelec NESDR SMArTee v2` with bundled telescoping antenna (~$50 CAD). Antenna length didn't seem to matter.

Run rtlamr in the background
```
docker run -d --name rtlamr -e LISTEN_ONLY=yes --device=/dev/bus/usb:/dev/bus/usb allangood/rtlamr2mqtt
```

Extract the logs to a file. Longest I ran a docker process was 22 hours and it kept all logs (~18MB).
```
docker logs --since 1000000m rtlamr &> rtlamr-2021-11-19.txt
```

Stats: Meter Ids with counts of number of messages
```
grep -o --no-filename -e '"EndpointID":[0-9]*' -e '"ID":[0-9]*' -e '"ERTSerialNumber":[0-9]*' rtlamr*.txt | sort | uniq -c
```

Stats: commodity types with count of number of messages. The commodities with a single message (like 157) I figure are noise that managed to pass the checksum.
```
$ grep -o --no-filename -e '"EndpointType":[0-9]*' -e '"Type":[0-9]*' -e '"ERTType":[0-9]*' rtlamr*.txt | sort | uniq -c

      1 "EndpointType":150
  28938 "EndpointType":156
      1 "EndpointType":157
      2 "EndpointType":170
  20181 "EndpointType":171
      1 "EndpointType":43
   2836 "ERTType":7
 178632 "Type":
      1 "Type":0
      1 "Type":10
  68757 "Type":12
  13001 "Type":13
     19 "Type":3
  41849 "Type":5
   3043 "Type":7
      1 "Type":9
```

Hope this helps others. I'm probably not going to continue this experiment. Was hoping to read my water meter but cannot. My gas meter isn't that interesting because there isn't much I can change to reduce useage and I already have natural gas detectors to notice leaks.

**Comment by ve6rah** [comment]
@brianlow My Enmax power meter was just replaced with a smart meter in Calgary.
It's not listed in the supported meters list for this project (marked "Honeywell Type RUD") however if you say you're seeing some in this area, I'm willing to try it out.
Thanks for posting this!
If you're giving up on the project, I may have a willing new home for your hardware...

**Reply by brianlow** [reply]
@ve6rah Lucky! Worth a try. My water meter isn't due for replacement for a few years. Yeah email me and we can work out a good deal if you want the hardware. My email is on my github profile.

**Reply by ve6rah** [reply]
Email sent. I had the meter replaced as part of a solar panel install, it was needed for bidirectional metering. While it doesn't have anything on it with an FCC ID, it does have a line that says "contains ic 4557a-rexuanz" which looks up as a 900 MHz transmitter chip.
I'm pretty sure my gas meter should be no issue as well, though I'll admit until I looked into reading the electric meter it hadn't even occurred to me to try the gas meter. As for water, I won't be able to read that one. Mine is not a smart meter at all, they still have to walk on to the property to read a display on the side of the house. They wanted to replace it a bunch of years ago with a smart meter, but the company they contracted to do it was unable to shut off the water at the street to do the work, and my meter is actually on the wrong side of the main water shut-off valve. I always assumed they would come back soon with a representative from the city to do it properly, but it never happened. I'm sure that meter is well past replacement age, I've lived here 21 years and it predates me, but it seems with age they're more likely to under report than over report, so I'm not sure that I should be the one in a rush to make it happen.

**Reply by vongyver** [reply]
A little more info for those interested on the Enmax Smart Meter in Calgary that I have discovered.  The Energy meter I has is a Honeywell Type RUD and 4556a-rexuanz you are both referring to.  It looks like it's made by Elster and Honeywell calls it a REXUniversal.  It run between 902.4 and 927.6 Mhz according to some references and reflects the vague documentation Honeywell has.  I am unable to determine the protocol it's running.  I added a line into meters.csv to reflect the configuration, playing with the ERT Type, but so far no luck.  I am not picking up any Electric meters, but plenty of gas and water.  I cannot find anything about what protocol the Elster units use. Some of my research suggests it may be LoRA based, so we may need an alternate system to pickup those signals. 

I am able to monitor my Water (Itron) and ATCO Gas Meter, I believe.  There are no clear markings per the IDs that I can find, so I narrowed it down with number of updates and comparing to the bills. The house was built in 2007.  I have also been able to correlate water heater and furnace heating demand to gas consumption and water usage to water consumption.

**Reply by ve6rah** [reply]
@vongyver I found the easiest way with the gas meter was to simply look at the display, and then look for one with the same values keeping in mind that the least significant digits are actually after the decimal point and therefore not shown on the meter's display. If you match the most significant digits though, it should narrow it down really quickly.
I would love to find the electric meter as well, the once my solar system is fully up and running with the right inverter I should have metering through it which would make the other one redundant.

**Comment by ve6rah** [comment]
Thankyou to @brianlow for setting me up with the hardware to do this!

Just got everything hooked up and am monitoring MQTT to find my meters. No luck so far on my electric meter overnight but I have seen a dozen gas meters, and a couple water meters. How often they transmit seems to really vary, one gas meter is transmitting every few minutes, others I've seen only once, mine varies between anywhere from 15 minutes to 2 and a quarter hours.

I'm sure my electric meter is transmission capable, I wonder if it's only on demand? if so, I should see it sometime next week on my normal read date. otherwise it's possible it's on a different frequency? Or even a different protocol? Not sure.

**Comment by brianlow** [comment]
I wonder if your power retailer or Enmax would be able to tell you more. I considered posting to a local group (like r/Calgary, FB, calgarypuck) in the hopes of finding someone that works closely with the meters. 

For gas, I wonder if some msgs are getting missed. Two hour interval seems like it would make automated meter reading hard. Don't know how to test the theory though. Though maybe didn't matter if gives you enough resolution for what you want. Does the interval vary by usage (nighttime = longer?)

**Comment by doctorkb** [comment]
If it helps any, up in Edmonton (Epcor electricity), the electricity meters use a dedicated Zigbee network -- in our case you need to get a Rainforest EAGLE-200 (or similar) and contact the utility to get it connected.

I'd reach out the Meter Shop to see - there's a phone number on this page: https://www.enmax.com/services/meter-services/meter-shop

**Comment by ve6rah** [comment]
The reason I don't think it's a zigbee network like Edmonton is that the chip advertised on the front of the meter is a 900 MHz transmitter, and zigbee works at 2.4 GHz.

It is possible that some messages are getting missed, in fact it seems likely. I'm just not entirely sure what to do about it, does not appear to be usage related as I didn't get any updates while I was using gas, but got 2 within 15 minutes when no gas was being used.

It's also possible that I'm just not able to receive my electric meter from my antenna's location, there's a bunch of concrete in the way, but the range seems good if I'm picking up a dozen gas meters, so who knows.

**Comment by brianlow** [comment]
Power: maybe try rtl_433 if you haven't already. I think it might decode more formats (and will listen on other frequencies on off chance that helps).

Gas: I wonder if there is a utility that provides signal strength? Make it easier to experiment with antenna orientation and length

**Comment by ve6rah** [comment]
My gas meter is pretty close to the SDR so I can't imagine I have a signal strength issue. It just doesn't broadcast on a reliable schedule.
here's what I've seen:
**22:56** 23:09 **00:27** 00:56 **01:44** 02:59 **03:14** 03:31 **05:43** 06:01 **08:33** 08:45 09:04 **09:27** **09:50** 09:52 **10:14** **10:25** 10:35 10:36 10:50 10:56 11:01 11:12 **11:51** 11:53 **12:03** 12:16 12:17 12:18 13:08 13:10 13:32 13:55 14:36 15:22 15:25 16:06 16:15 16:32 16:42 **17:16** 17:42 17:49 18:01 18:09 18:18 **19:36** 19:38 19:41 **19:48** 20:03 20:27 **21:17** **22:06** 22:09 22:23

There are gaps in there as short as a single minute, and ones as long as 2.5 hours. the consumption value only changed on the highlighted times, so doesn't seem related to that either.

**Comment by brianlow** [comment]
Maybe it is an anti-collision mechanism so all meters don't transmit at the same time

**Comment by vongyver** [comment]
You may want to look at some smart energy monitoring solutions. I use smart
monitoring for power monitoring and control and graph in Home Assistant. I
am able to simulate a renewable system with solar, wind and batteries which
will help in future planning. If you aren't already, and have some
programming skills or willing to learn, there are loads of great sources
for building blocks.

**Comment by mcattle** [comment]
This has been a great help in importing my water and gas meter data into Home Assistant (in Calgary) using [rtlamr2mqtt](https://github.com/allangood/rtlamr2mqtt).  I haven't been able to read my two-way electric meter, but I have found this spec sheet if it helps someone with more knowledge of SDR than me:  [4557a-rexuanz Manual](https://fccid.io/QZC-REXUANZ/User-Manual/Manual-2738132.pdf)

Both my water and gas meters fortunately had their digital ID printed on the meter, so they were easy to identify.  (House built in 2013.)

**Reply by doctorkb** [reply]
In Edmonton, the meters that are used have some form of two-way communication with Epcor.  This isn't a simple "pick up the signal as it's being sent" that our gas (Atco) and older water meters have.

In looking at Enmax's info, 90% of the electricity meters in Calgary are being read manually (that is, not even by picking up the RF signal, but actually walking up to the meter).  Are you certain that yours is read electronically?

**Reply by mcattle** [reply]
Yes, it's a newer two-way smart meter with a comms chip.

**Reply by doctorkb** [reply]
That's great that it is capable of it, but doesn't answer whether it is activated.

I also suspect it is like Epcor's meters that don't sent a pulse to be picked up, but have a network connection. If that's the case, you'd need to get their assistance in order to get real-time readings... Or set up an AI-on-the-edge camera to read it visually.

**Reply by joryirving** [reply]
I'm trying to setup rltamr2mqtt with my honeywell meter in Calgary. Which ID am I using? `LAN ID`? (House built in 1982)

**Reply by ve6rah** [reply]
@joryirving You're going to have to be more specific. Which utility are you talking about? Water, or gas? (I don't think you can get electric this way, they appear to be a different technology, if they're connected at all, which most of them are not)
My gas meter is itron ink, and has a big barcode on the top with a big bold number. That's the one to feed into this. My water meter is also itron and has a large bold number printed on the top of it as well.

**Reply by joryirving** [reply]
It's the honeywell electricity meter, which was mentioned above, but re-reading, I see that you're right, it may need another system. I still need to physically find my water/gas meters and see if I can import those.

**Reply by ve6rah** [reply]
@joryirving I would be surprised if your electric meter has been updated to one of the new ones with communication, the vast majority of ones in Calgary are still analog and manual read. If it is digital, there are ways of reading it, but I don't think you're going to find this integration helpful. It seems to me that your best bet for reading one of these is going to be one of the optical methods.
For gas and water though, most ones in the city have been updated, and are easy enough to read. For gas the meter will be on the outside of the house and should be easy enough to check. For water the meter itself is going to be inside your house, but the piece you actually want to look at will be run to the outside. It'll be a small gray box with no display, and will not specify that it is a water meter. Mine is mounted by a furnace air intake vent, because that's how they ran the wire, yours will also likely be mounted at some other wall penetration. 
The gas meter seems to update pretty much in real time when reading it, the water meter I find a little more odd, as it broadcasts on a very frequent basis, but only actually updates hourly. So it makes it hard to check real-time consumption.

**Reply by joryirving** [reply]
I mean, I've confirmed it's definitely a Honeywell unit with wireless communications, as mentioned here: https://github.com/bemasher/rtlamr/discussions/217#discussioncomment-3544777

But it sounds like that won't work with the sdr_rtl module I have.

**Comment by ve6rah** [comment]
> In looking at Enmax's info, 90% of the electricity meters in Calgary are being read manually (that is, not even by picking up the RF signal, but actually walking up to the meter). Are you certain that yours is read electronically?

You are correct that most meters in Calgary are still the old-fashioned style with manual reads. That said when I installed my solar system they installed a bi-directional meter which is fully electronic and able to be read remotely. That said, I'm not sure how feasible it is for us to read them. In my case it is been low priority because my solar system already provides the same information, so I've mostly just been using this for gas and water.
